packages/eslint-plugin/src/rules/consistent-type-definitions.ts

```diff
@@ -31,6 +32,21 @@ export default util.createRule({
   create(context, [option]) {
     const sourceCode = context.getSourceCode();
 
+    function isNodeGrandparentGlobalModuleDeclaration(
+      node: TSESTree.Node,
+    ): boolean {
+      if (
+        node.parent?.type === AST_NODE_TYPES.TSModuleBlock &&
+        node.parent.parent?.type === AST_NODE_TYPES.TSModuleDeclaration &&
+        node.parent.parent?.declare &&
+        node.parent.parent?.global
```

Comment:
instead of just looking at the grandparent - we could instead look at all the ancestors so we can catch cases like:
```ts
declare global {
  namespace Foo {
    interface Bar {}
  }
}
```

`context.getAncestors()` will return you an array and you can traverse it from the start to go from the highest parent.